omod/src/main/webapp/resources/scripts/directives/select-drug.js

```diff
@@ -16,7 +16,9 @@ angular.module('uicommons.widget.select-drug', [ 'drugService', 'ui.bootstrap' ]
                 $scope.size = attrs.size ? attrs.size : 40;
 
                 $scope.search = function(term) {
-                    return DrugService.getDrugs({ q: term });
+                    return DrugService.getDrugs({ q: term }).then(function(result){
```

Comment:
Did you test this to ensure it works? I thought `getDrugs` returns the actual results and not a promise as seen [here](https://github.com/openmrs/openmrs-module-uicommons/blob/master/omod/src/main/webapp/resources/scripts/services/drugService.js#L20)